pycalphad/core/composition_set.pyx

```diff
@@ -35,18 +36,32 @@ cdef public class CompositionSet(object)[type CompositionSetType, object Composi
         other._energy_2d_view = <double[:1]>&other.energy
         other.NP = 1.0*self.NP
         other.fixed = bool(self.fixed)
+        other.phase_local_cons_func = self.phase_local_cons_func
+        other.phase_local_cons_jac = self.phase_local_cons_jac
+        other.num_phase_local_conditions = int(self.num_phase_local_conditions)
         return other
 
     def __repr__(self):
         return str(self.__class__.__name__) + "({0}, {1}, NP={2}, GM={3})".format(self.phase_record.phase_name,
                                                                           np.asarray(self.X), self.NP, self.energy)
 
+    cpdef void set_local_conditions(self, dict phase_local_conditions):
+        mod = self.phase_record.phase_record_factory.models[self.phase_record.phase_name]
+        cfuncs = build_phase_local_constraints(mod, self.phase_record.state_variables + self.phase_record.variables,
+                                               phase_local_conditions,
+                                               parameters=self.phase_record.phase_record_factory.param_symbols)
+        self.phase_local_cons_func = FastFunction(cfuncs.internal_cons_func)
+        self.phase_local_cons_jac = FastFunction(cfuncs.internal_cons_jac)
+        self.num_phase_local_conditions = cfuncs.num_internal_cons
+
```

Comment:
Maybe I'm not understanding, but this function seems to not actually do anything? Related `self.phase_local_cons_func`/`self.phase_local_cons_func` not being read anywhere.

Reply:
`compset.phase_local_cons_jac` is used on minimizer.pyx:65. `PhaseRecord.phase_local_cons_{func,jac}` accepts a `FastFunction` as an argument, and this object is stored on the `CompositionSet`.